pysr/regressor_sequence.py

```diff
@@ -0,0 +1,299 @@
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0t_1", "x1t_2", etc.
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        complexity_of_variables : int | float | list[int] | list[float]
+            The complexity of each variable in `X`. If a single value is
+            passed, it will be used for all variables. If a list is passed,
+            its length must be the same as `recurrence_history_length`.
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        X = self._validate_data(X)
+        _check_assertions(
+            X,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+        )
+        self.variable_names = variable_names  # for latex_table()
+        self.n_features = X.shape[1]  # for latex_table()
+
+        current_X = X[self.recursive_history_length :]
+        historical_X = self._sliding_window(X)[: -1 : current_X.shape[1], :]
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+        variable_names = self._construct_variable_names(
+            current_X.shape[1], variable_names
+        )
+
+        self._regressor.fit(
+            X=historical_X,
+            y=current_X,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+        return self
+
+    def predict(self, X, index=None, num_predictions=1):
+        """
+        Predict future data from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Data of shape `(n_times, n_features)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+        num_predictions : int = 1
```

Comment:
```suggestion
        num_predictions : int
```

Same format as elsewhere